plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java

```diff
@@ -531,6 +547,16 @@ public String getEffectivePom(String projectPath) {
     return result;
   }
 
+  private void reComputeDiagnostics(String pomPath) {
+    String pomUri = prefixURI(pomPath);
+    Type type = new TypeToken<PublishDiagnosticsParams>() {}.getType();
+    PublishDiagnosticsParams result =
+        doGetOne(Commands.RECOMPUTE_POM_DIAGNOSTICS, singletonList(pomUri), type);
+    CheLanguageClient cheLanguageClient =
```

Comment:
Don't create `CheLanguageClient`

It is enough to add a new method:
```
  public void publishDiagnostics(PublishDiagnosticsParams diagnostics) {
    eventService.publish(new ExtendedPublishDiagnosticsParams(serverId, diagnostics));
  }
```

Reply:
fixed